hyperspy/drawing/markers.py

```diff
@@ -0,0 +1,774 @@
+    in the plot.
+
+    For example if we wanted to define a set of ellipses with constant
+    height, width, and size we can do the following.
+
+    >>>from matplotlib.collections import EllipseCollection
+    >>>import hyperspy.api as hs
+    >>>import hyperspy.api as hs
+    >>>import numpy as np
+    >>>m = hs.plot.marker.Markers(collection_class=EllipseCollection, widths=(2,),
+    >>>                 heights=(1,), angles=(1,), units="xy", offsets=np.random.rand(10,2)*10)
+    >>>s = hs.signals.Signal2D(np.ones((10,10,10,10)))
+    >>>s.plot()
+    >>>s.add_marker(m)
+
+    To define a non-static marker any kwarg that can be set with the `matplotlib.collections.set`
+    method can be passed as an array with `dtype=object` and the same size as the navigation axes
+    for a signal.
+    """
+    marker_type = "Markers"
+
+    def __init__(self,
+                 collection_class,
+                 offset_units="data",
+                 transform_units=None,
+                 size_units=None,
+                 shift=None,
+                 **kwargs):
+        """
+        Initialize a Marker Collection.
+
+        Parameters
+        ----------
+        collection_class : matplotlib.collections or str
+            A Matplotlib collection to be initialized.
+        offsets : [n, 2]
+            Positions of the markers
+        offset_units: str
+            The units of the offsets can be one of the following:
+            "data": the offsets are in data coordinates or real units
+            "identity": the offset is not transformed
+            "axes": the offsets are in axes coordinates (0-1) and 0, 0) is the bottom left of your axes
+             or subplot, (0.5, 0.5) is the center, and (1.0, 1.0) is the top right
+            "yaxis":The offsets are in data coordinates in y and axes coordinates in x
+            "xaxis": The offsets are in data coordinates in x and axes coordinates in y
+            "rel": The offsets are in data coordinates in x and coordinates in y relative to the
+            data plotted.
+        transform_units: str or None
+            The units of the transform can be one of the following:
+            "data": the transform is in data coordinates or real units
+            "axes": the transform is in axes coordinates (0-1) and 0, 0) is the bottom left of your axes
+            or subplot, (0.5, 0.5) is the center, and (1.0, 1.0) is the top right
+            "yaxis":The transform is in data coordinates in y and axes coordinates in x
+            "xaxis": The transform is in data coordinates in x and axes coordinates in y
+            "rel": The transform is in data coordinates in x and coordinates in y relative to the
+            data plotted.
+        size_units: str or None
+            The units of the size can be one of the following:
+            "yaxis": The size is based on the scale for the y axis
+            "xaxis": The size is based on the scale for the x axis
```

Comment:
What's about mapping directly to the matplotlib collections arguments? Do we need the `size_units` argument?

```suggestion
        offsets_transform: str
            Define the transformation used for the `offsets`. It can be one of the following:
            - ``"data"``: the offsets are defined in data coordinates and the ``ax.transData`` transformation is used.
            - ``"relative"``: The offsets are defined in data coordinates in x and coordinates in y relative to the
              data plotted. Only for 1D figure.
            - ``"axes"``: the offsets are defined in axes coordinates and the ``ax.transAxes`` transformation is used.
              (0, 0) is bottom left of the axes, and (1, 1) is top right of the axes.
            - ``"xaxis"``: The offsets are defined in data coordinates in x and axes coordinates in y direction; use
              :py:meth:`matplotlib.axes.Axes.get_xaxis_transform` transformation.
            - ``"yaxis"``: The offsets are defined in data coordinates in y and axes coordinates in x direction; use
              :py:meth:`matplotlib.axes.Axes.get_xaxis_transform` transformation..
            - ``"display"``: the offsets are not transformed, i.e. are defined in the display coordinate system.
              (0, 0) is the bottom left of the window, and (width, height) is top right of the output in "display units"

        transform: str or None
            Define the transformation used for the `offsets`. It can be one of the following:
            ... same as above
```

Reply:
The thing we might not need is a separate offset_transform and transform.  The only time this is useful is when you are trying to define a bunch of lines or shapes by `segments` or by `verts`

Reply:
Maybe for now, we simply set some sensible default when the collection is initiated and as you said above, user can always change it the `kwargs` or using `m.collection.set_transform(m.ax.transData)` or `m.collection.set_offset_transform(m.ax.transData)`. With some example in the gallery, this can be documented and use as inspiration.
We may be overthinking it and later when we can add more functionality: it is easier to add API than going through trough a deprecation cycle.